nim/tests/test_routes.py

```diff
@@ -42,14 +59,36 @@ def assert_new_game_success(client, request):
     assert max(state) <= request.get('max', DEFAULT_MAX) <= NUM_LIMIT_MAX
 
 
-def assert_new_game_failed(client, request):
+@pytest.mark.parametrize("request", [
```

Comment:
Oh man, this is a hell of a change. 
Did you always have this implementation in mind or did you think of it after I pushed my stuff? 
It's honestly beautiful. Not only does it save a lot of lines, but it makes the tests so much clearer and neater with less repetition. I love it
There's a lot I can learn from this